Sources/FrontEnd/Parse/Lexer.swift

```diff
@@ -322,10 +343,11 @@ public struct Lexer: IteratorProtocol, Sequence {
     return sourceCode.text[start ..< index]
   }
 
-  /// Consumes a numeric literal and returns its kind, or returns `nil` if the stream starts with a
-  /// different token.
-  private mutating func scanNumericLiteral() -> Token.Kind? {
-    let i = take("-") ?? index
+  /// Consumes an integral literal and returns its kind, or returns `nil` if it fails to scan a valid integer.
+  /// - Parameter allowingPlus: If set to `true`, allows the integral literal to begin with a "+" sign.
```

Comment:
```suggestion
  ///
  /// - Parameter allowingPlus: If set to `true`, allows the integral literal to begin with a "+" sign.
```
Nice comment! I would not have guessed the behavior if `false`.